gateway/src/apicast/mapping_rules_matcher.lua

```diff
@@ -52,4 +52,21 @@ function _M.matches(method, uri, args, rules)
   return false
 end
 
+function _M.clean_usage_by_owner_id(rules, owner_id)
+  local usage = Usage.new()
+  if not owner_id then
+      return usage
+  end
+
+  for _, rule in ipairs(rules) do
+    -- Checking that the rule has owner_id, if it does not have owner_id means
+    -- that does not belongs to a backendAPI.
+    if rule.owner_id and owner_id ~= rule.owner_id then
+      usage:add(rule.system_name, 0 - (rule.delta or 0))
```

Comment:
Nitpick: I think the first 0 is redundant.

Reply:
it's clear that we are substracting that, I would keep to make it clear.